app/Http/Controllers/AmaController.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\DB;
+
+class AmaController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index(Request $request)
+    {
+       
+   
+         $amas = DB::table('amas')->select('title', 'person', 'tags', 'created_at')->paginate(20);
+
+        
+        echo $request->date_from;
+        echo $request->date_to;
+        echo $request->tags;
+        
+            if ($request->has('date_from')) {
+
+
+               $amas = DB::table('amas')->select('title', 'person', 'tags', 'created_at')->whereBetween('created_at', [$request->date_from, $request->date_to])->paginate(20);
+
+
+                }     
+
+         
+             if(request()->has('tags')){
+                $amas = DB::table('amas')->select('title', 'person', 'tags', 'created_at')->where('tags' , $request->tags )->paginate(20);
+                }
+
+        
+
+        
+          return view('ama_list',['amas' => $amas]);
```

Comment:
After doing the changes above, don't forget to do `$amas = $amas->paginate(20)`.